### PR TITLE
[build-script] Pass updated arguments to SwiftSyntax's build script

### PR DESCRIPTION
**Pull request by ahoppen**
Update the parameters passed to SwiftSyntax's build script to fix an issue that caused SwiftSyntax's module to not be properly installed.

Corresponds to https://github.com/apple/swift-syntax/pull/151

### CODE REVIEW

**Comment by ahoppen** [comment]
https://github.com/apple/swift-syntax/pull/151

@swift-ci Please test

File: utils/build-script-impl [hunk 1]
```diff
@@ -3443,7 +3443,7 @@ for host in "${ALL_HOSTS[@]}"; do
                 if [ "${BUILD_LIBPARSER_ONLY}" ]; then
                     # We don't have a toolchain so we should install to the specified dir
                     DYLIB_DIR="${INSTALL_DESTDIR}"
-                    MODULE_DIR="${INSTALL_DESTDIR}"
+                    MODULE_DIR="${INSTALL_DESTDIR}/SwiftSyntax.swiftmodule/${SWIFT_HOST_VARIANT_ARCH}"
```
Comment (nkcsgexi) [inline utils/build-script-impl]: Can we use `${product}` for `SwiftSyntax` here?

Comment (ahoppen) [reply utils/build-script-impl]: Good idea. Changed it.

**Comment by ahoppen** [comment]
https://github.com/apple/swift-syntax/pull/151

@swift-ci Please test

**Comment by gottesmm** [comment]
@ahoppen question. Can you try using the build-script toolchain approach? Rather than using build-script-impl?

**Comment by gottesmm** [comment]
nm, I see you are actually doing that! Big +1 from me!

**Pull request merged**